From 8d2caa61aa9f0d9a38733e9d11a9b943b19455c5 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Sat, 22 Aug 2020 21:18:48 +0530
Subject: [PATCH 627/767] octeontx2-af: Fix the BPID mask

commit 72d2a3acac11ded56cf2ac212fae246a811f1871 from
git@git.assembla.com:cavium/WindRiver.linux.git

The backpressure ID to be programmed in
NIX_AF_RX_CHAN(0..4095)_CFG has bit definition
as [8:0] but the mask used to configure is 0xFF
by mistake. This patch fixes that.

Fixes: df6e0876b1e8
("octeontx2-af: Add mbox messages to configure
backpressure for an interface")

Change-Id: I3949ca78dcae83e0a63d57292c6b370f25dabf2d
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/34447
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
index 589f1679f03b..022b0108f342 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
@@ -455,8 +455,9 @@ int rvu_mbox_handler_nix_bp_enable(struct rvu *rvu,
 		}
 
 		cfg = rvu_read64(rvu, blkaddr, NIX_AF_RX_CHANX_CFG(chan));
+		cfg &= ~GENMASK_ULL(8, 0);
 		rvu_write64(rvu, blkaddr, NIX_AF_RX_CHANX_CFG(chan),
-			    cfg | (bpid & 0xFF) | BIT_ULL(16));
+			    cfg | (bpid & GENMASK_ULL(8, 0)) | BIT_ULL(16));
 		chan_id++;
 		bpid = rvu_nix_get_bpid(rvu, req, type, chan_id);
 	}
-- 
2.31.1

